src/client/serialize/hdf_serialize.c

```diff
@@ -0,0 +1,1034 @@
+#include <stdio.h>
+#include <hdf5.h>
+#include <daos.h>
+#include <daos_cont.h>
+
+/* for user attr dataset */
+struct usr_attr {
+	char	*attr_name;
+	hvl_t	attr_val;
+};
+
+static int
+cont_serialize_prop_acl(hid_t *file_id, struct daos_prop_entry *entry,
+			const char *prop_str)
+{
+
+	int		rc = 0;
+	hid_t		status = 0;
+	struct daos_acl	*acl = NULL;
+	char		**acl_strs = NULL;
+	size_t		len_acl = 0;
+	hsize_t		attr_dims[1];
+	hid_t		attr_dtype;
+	hid_t		attr_dspace;
+	hid_t		usr_attr;
+
+
+	if (entry == NULL || entry->dpe_val_ptr == NULL) {
+		goto out;
+	}
+
+	/* convert acl to list of strings */
+	acl = (struct daos_acl *)entry->dpe_val_ptr;
+	rc = daos_acl_to_strs(acl, &acl_strs, &len_acl);
+	if (rc != 0) {
+		fprintf(stderr, "failed to convert acl to strs");
+		goto out;
+	}
+	attr_dims[0] = len_acl;
+	attr_dtype = H5Tcopy(H5T_C_S1);
+	if (attr_dtype < 0) {
+		fprintf(stderr, "failed to create acl type");
+		rc = 1;
+		goto out;
```

Comment:
Could we use D_GOTO in this file?
```suggestion
		D_GOTO(out, rc = 1)
```